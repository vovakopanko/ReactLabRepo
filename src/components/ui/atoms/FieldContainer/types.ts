export type FieldType = {
  id?: number;
  title: string | number;
  titleName: string;
};

export type Props = {
  id?: number;
  title: string | number;
  titleName: string;
};
