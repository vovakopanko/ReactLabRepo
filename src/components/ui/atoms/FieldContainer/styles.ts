import styled from "styled-components";
import { colors } from "./../../../../styles/palette/index";
export const SettingInfoBlock = styled.div`
  display: flex;
  align-items: "center";
  flex-direction: "row";
  border-bottom: 1px solid ${colors.BLACK};
  &:last-child{
    border-bottom: none };
  }
`;

export const SettingLabel = styled.p`
  width: 50%;
  color: ${colors.GRAY};
  font-size: 18px;
  padding-left: 5px;
`;

export const SettingText = styled.span`
  text-align: right;
  word-wrap: break-word;
  align-self: center;
  width: 50%;
  color: ${colors.BLACK};
  font-size: 20px;
`;
