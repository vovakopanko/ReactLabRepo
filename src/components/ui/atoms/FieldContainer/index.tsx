import { SettingInfoBlock, SettingLabel, SettingText } from "./styles";
import { Props } from "./types";

const FieldContainer = ({ title, titleName }: Props) => {
  const titleLength = titleName.substring(0, 20);
  return (
    <SettingInfoBlock>
      <SettingLabel>{titleLength}</SettingLabel>
      <SettingText>{title && title}</SettingText>
    </SettingInfoBlock>
  );
};

export default FieldContainer;
