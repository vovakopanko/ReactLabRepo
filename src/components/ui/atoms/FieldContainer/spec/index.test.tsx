import renderer from "react-test-renderer";
import FieldContainer from "..";

describe("FieldContainer component unit test", () => {
  it("should match to snapshot", () => {
    const tree = renderer
      .create(<FieldContainer title={"Title"} titleName={"Description"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
