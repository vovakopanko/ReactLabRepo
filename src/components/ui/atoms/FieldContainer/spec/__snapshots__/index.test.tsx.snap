// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldContainer component unit test should match to snapshot 1`] = `
<div
  className="sc-bdvvtL jfhSOZ"
>
  <p
    className="sc-gsDKAQ kfPKVl"
  >
    Description
  </p>
  <span
    className="sc-dkPtRN dIidvx"
  >
    Title
  </span>
</div>
`;
